api/src/main/java/ca/bc/gov/educ/studentdatacollection/api/service/v1/SdcDuplicatesService.java

```diff
@@ -116,20 +117,23 @@ public void checkIfDuplicateIsGeneratedAndThrow(SdcSchoolCollectionStudentEntity
       List<SdcSchoolCollectionStudentEntity> allStudentsWithSameAssignedStudentId = sdcSchoolCollectionStudentRepository
               .findAllDuplicateStudentsByCollectionID(sdcSchoolCollectionStudentEntity.getSdcSchoolCollection().getCollectionEntity().getCollectionID(), Collections.singletonList(sdcSchoolCollectionStudentEntity.getAssignedStudentId()));
 
-      List<SdcSchoolCollectionStudentLightEntity> duplicateStudentEntities = new ArrayList<>();
+      //filter out the non-updated entity fetched from db
+      List<SdcSchoolCollectionStudentEntity> filteredStudentsWithSameAssignedStudentId = allStudentsWithSameAssignedStudentId.stream().filter(student -> !student.getSdcSchoolCollectionStudentID().equals(sdcSchoolCollectionStudentEntity.getSdcSchoolCollectionStudentID())).toList();
+
       //map to light object
-      duplicateStudentEntities.addAll(allStudentsWithSameAssignedStudentId.stream().map(sdcSchoolCollectionStudentMapper::toSdcSchoolStudentLightEntity).toList());
-      if (sdcSchoolCollectionStudentEntity.getSdcSchoolCollectionStudentID() == null) {
-        duplicateStudentEntities.add(sdcSchoolCollectionStudentMapper.toSdcSchoolStudentLightEntity(sdcSchoolCollectionStudentEntity));
-      }
+      List<SdcSchoolCollectionStudentLightEntity> duplicateStudentEntities = new ArrayList<>(filteredStudentsWithSameAssignedStudentId.stream().map(sdcSchoolCollectionStudentMapper::toSdcSchoolStudentLightEntity).toList());
+      duplicateStudentEntities.add(sdcSchoolCollectionStudentMapper.toSdcSchoolStudentLightEntity(sdcSchoolCollectionStudentEntity));
```

Comment:
What if this is an existing student during an update? They would be returned in the set on line 117 and we're adding them here again. I think we need the null check.